File: 3618/3618.c
int splitArrayByPrimeIndices(struct TreeNode* root) {
    if (!root) return 0;
    int l = splitArrayByPrimeIndices(root->left);
    int r = splitArrayByPrimeIndices(root->right);
    return 1 + (l > r ? l : r);
}

File: 3618/3618.go
func splitArrayByPrimeIndices(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := splitArrayByPrimeIndices(root.Left)
	r := splitArrayByPrimeIndices(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}